core/src/main/java/feign/Feign.java

```diff
@@ -140,6 +140,10 @@ public Builder decoder(Decoder decoder) {
      * {@link Util#emptyValueOf}. To customize further, wrap an existing
      * {@link #decoder(Decoder) decoder} or make your own.
      *
+     * In order to allow {@link Decoder}s to delegate error handling to {@link ErrorDecoder}s,
```

Comment:
I appreciate the thought, but I don't think this will reduce confusion. There's no introduction of what wrapping exceptions is about.

If anything, we'd add a section in the javadoc of Decoder about exception handing, mentioning that runtime exceptions are wrapped as DecodeExceptions except when decode404 is set. I hate that this is even possible, so really probably just put off about us adding the complexity of allowing this.